src/analyses/apron/apronAnalysis.apron.ml

```diff
@@ -139,17 +152,35 @@ struct
       apr
 
 
+  let replace_deref_exps ask e =
+    let rec inner e = match e with
+      | Const x -> e
+      | UnOp (unop, e, typ) -> UnOp(unop, inner e, typ)
+      | BinOp (binop, e1, e2, typ) -> BinOp (binop, inner e1, inner e2, typ)
+      | CastE (t,e) -> CastE (t, inner e)
+      | Lval (Var v, off) -> Lval (Var v, off)
+      | Lval (Mem e, NoOffset) ->
+        (match ask (Queries.MayPointTo e) with
+         | (`Lifted _) as r when (Queries.LS.cardinal r) = 1 ->
+           let lval = Lval.CilLval.to_lval (Queries.LS.choose r) in
+           Lval lval
+         | _ -> Lval (Mem e, NoOffset))
```

Comment:
Unlike the `MayPointTo` in `assign_to_global_wrapper`, this doesn't perform a join over all possible addresses/values. Not sure it matters for anything real, but if all pointed to values are equal according to Apron (or their join could be bounded, etc), then there's extra information that currently wouldn't be used.

Getting it to work would be significantly more involved though. I guess one would need something like `read_globals_to_locals` which reads (and joins) all dereferenced things into temporary locals and then computes with those instead.

Reply:
True! For now, let's skip this and add it if we notice it has significant performance penalties.

Reply:
Maybe add a comment/TODO about it here (possibly with a comment link).

Reply:
:+1: